lib/saluki-env/src/autodiscovery/providers/local.rs

```diff
@@ -0,0 +1,298 @@
+use std::path::PathBuf;
+
+use async_trait::async_trait;
+use saluki_error::GenericError;
+use tokio::fs;
+use tokio::sync::broadcast::{self, Receiver, Sender};
+use tokio::sync::OnceCell;
+use tokio::time::{interval, Duration};
+use tracing::{debug, warn};
+
+use crate::autodiscovery::{AutodiscoveryEvent, AutodiscoveryProvider, Config, EventType};
+
+/// A local auto-discovery provider that uses the file system.
+pub struct LocalAutoDiscoveryProvider {
+    search_paths: Vec<PathBuf>,
+    sender: Sender<AutodiscoveryEvent>,
+    listener_init: OnceCell<()>,
+}
+
+impl LocalAutoDiscoveryProvider {
+    /// Creates a new `LocalAutoDiscoveryProvider` that will monitor the specified paths.
+    pub fn new<P: AsRef<std::path::Path>>(paths: Vec<P>) -> Self {
+        let search_paths: Vec<PathBuf> = paths
+            .iter()
+            .filter_map(|p| {
+                if !p.as_ref().exists() {
+                    warn!("Skipping path '{}' as it does not exist", p.as_ref().display());
+                    return None;
+                }
+                if !p.as_ref().is_dir() {
+                    warn!("Skipping path '{}', it is not a directory.", p.as_ref().display());
+                    return None;
+                }
+                Some(p.as_ref().to_path_buf())
+            })
+            .collect();
+
+        let (sender, _) = broadcast::channel::<AutodiscoveryEvent>(100);
+
+        Self {
+            search_paths,
+            sender,
+            listener_init: OnceCell::new(),
+        }
+    }
+
+    /// Starts a background task that periodically scans for configuration changes
+    async fn start_background_monitor(&self, interval_sec: u64) {
+        let mut interval = interval(Duration::from_secs(interval_sec));
+        let sender = self.sender.clone();
+        let search_paths = self.search_paths.clone();
+
+        tokio::spawn(async move {
+            let mut known_configs = HashSet::new();
+
+            loop {
+                interval.tick().await;
+
+                // Scan for configurations and emit events for changes
+                if let Err(e) = scan_and_emit_events(&search_paths, &mut known_configs, &sender).await {
+                    warn!("Error scanning for configurations: {}", e);
+                }
+            }
+        });
+    }
+}
+
+/// Parse a YAML file into a Config object
+async fn parse_config_file(path: &PathBuf) -> Result<(String, Config), GenericError> {
+    let content = fs::read_to_string(path).await?;
+
+    // Build config ID from the file path
+    let config_id = path
+        .canonicalize()?
+        .to_string_lossy()
+        .to_string()
+        .replace(['/', '\\'], "_");
+
+    // Create a Config
+    let config = Config {
+        name: path.file_name().unwrap().to_string_lossy().to_string(),
+        event_type: EventType::Schedule,
+        init_config: content.as_bytes().to_vec(),
+        instances: Vec::new(), // Would parse from YAML in a real implementation
```

Comment:
Is there any reason we shouldn't make it actually read from YAML as part of this PR?

Reply:
I parsed the ones that we care for checks `init_config` and `instances`. I think we can leave for now like this and the more we use other fields we can add the parsig bit in the local implmentation